b2luigi/batch/processes/htcondor.py

```diff
@@ -0,0 +1,225 @@
+import json
+import logging
+import os
+import re
+import shutil
+import stat
+import subprocess
+import enum
+import sys
+
+from b2luigi.core.utils import get_setting, get_output_dirs
+from b2luigi.batch.processes import BatchProcess, JobStatus, BatchJobStatusCache
+from b2luigi.core.utils import get_log_file_dir
+
+
+class HTCondorJobStatusCache(BatchJobStatusCache):
+
+    def __init__(self):
+        super(HTCondorJobStatusCache, self).__init__()
+
+    def _ask_for_job_status(self, job_id: int = None):
+        q_cmd = ["condor_q", "-json", "-attributes", "ClusterId,ProcId,JobStatus"]
+        history_cmd = ["condor_history", "-json", "-attributes", "ClusterId,ProcId,JobStatus"]
+
+        if job_id:
+            q_cmd.append(str(job_id))
+            output = subprocess.check_output(q_cmd)
+        else:
+            output = subprocess.check_output(q_cmd)
+        output = output.decode()
+
+        dict_list = []
+        for status_dict in output.lstrip("[\n").rstrip("\n]\n").split("\n,\n"):
+            if status_dict:
+                dict_list.append(json.loads(status_dict))
+            else:
+                continue
+
+        if job_id:
+            if job_id not in [status_dict["ClusterId"] for status_dict in dict_list]:
+                history_cmd.append(str(job_id))
+                output = subprocess.check_output(history_cmd)
+                output = output.decode()
+                output = output.lstrip("[\n").rstrip("\n]\n").split("\n,\n")
+                if output:
+                    for status_dict in output:
+                        dict_list.append(json.loads(status_dict))
+
+        for status_dict in dict_list:
+            self[status_dict["ClusterId"]] = status_dict["JobStatus"]  # int -> int
```

Comment:
I do not understand this comment

Reply:
It"s just a reminder that both, the key and the value in this dictionary are Integers. When i
extracted the `job_id` after the submission via a search using a regular expression, the `job_id` was given as string (see line 133). Checking the status in the cache didn't work back then, so i decided to cast the `job_id` to a `int`.